fix: add a reference to eventing sources page instead of having a list (#2939)
knative/docs: docs/eventing/samples/writing-event-source/README.md

Just want to see the code? The reference project is
[https://github.com/knative-sandbox/sample-source](https://github.com/knative-sandbox/sample-source).

> [Knative Sources](../../sources/_index.md#knative-sources) can be used as a reference.

Reviewer: ```suggestion
[Knative Sources](../../sources/#knative-sources) can be used as a reference.
```

All `_file.ext` are internal build files and should not be directly linked to. This should fix this (note that this is not the convention the rest of the library uses). See comment below for more context.

Author: yeah, I was actually wondering about that! 
Thanks for the suggestion

> [Knative Sources](https://knative.dev/docs/eventing/sources/#knative-sources) can be used as a reference.

Reviewer 2: For content in the Knative site, please use relative md links instead of URLs, example: https://github.com/knative/docs/pull/2703/files

> A variety of event sources are available in Knative [`eventing-sources-page`](https://knative.dev/docs/eventing/sources/#knative-sources)

Reviewer 3: ```suggestion
[Knative Sources](https://knative.dev/docs/eventing/sources/#knative-sources) can be used as a reference.
```
Does this read better? :thinking: 

/cc mpetason abrennan89

> is an easy way to turn any dispatcher container into an Event Source. Similarly, another option is using [SinkBinding](../../../eventing/sources/README.md#meta-sources)

Reviewer: @abrennan89  It looks like someone removed the actual README file without cleaning up all the links here (and throughout the library?).

This continues to work because the build script strips the `/README.md` filename and extension (which is part of whats causing the confusion). BUT.. we need to pick one way, consistently apply those changes, and backport the library because we cant change the build script if we are doing two different things.
